File: src/main/java/com/tavisca/workshops/pratham/rover/Rover.java
package com.tavisca.workshops.pratham.rover;

import com.tavisca.workshops.pratham.rover.model.Command;
import com.tavisca.workshops.pratham.rover.model.Vector;

public class Rover {
    private Vector vector;

    public Rover(Vector initialVector) {
        vector = initialVector;
    }

    public Vector rove(Command command) {
        switch (command){
            case TURN_LEFT:
                vector = vector.left();
                break;
            case TURN_RIGHT:
                vector = vector.right();
                break;
            case MOVE_FORWARD:
                vector = vector.next();
                break;
        }
        return vector;
    }

    public Vector vector(){
        return vector;
    }
}

File: MarsRover/Rover.cs
using System;

namespace MarsRover
{
    public class Rover
    {
        private Coordinate _currentCoordinate;
        public Direction _currentDirection;
        private Plateau _plateau;

        public Rover(Plateau plateau, Direction currentDirection, Coordinate currentCoordinate)
        {
            _currentCoordinate = currentCoordinate;
            _plateau = plateau;
            _currentDirection = currentDirection;
        }

        public void Run(string CommandString)
        {
            var RoverCommands = new CommandProcessor(CommandString).ToCommands();
            foreach(var command in RoverCommands)
            {
                command.Execute(this);
            }
        }

        public string GetCurrentLocation()
        {
            return _currentCoordinate.ToString().Trim() + " " + _currentDirection.ToString().Trim();
        }



        public void Move()
        {
            var PositionAfterMove = _currentCoordinate.NewCoordinatesForStepSize(_currentDirection.XAxisStepSize, _currentDirection.YAxisStepSize);
            
            if (_plateau.HasWithinBounds(PositionAfterMove))
                _currentCoordinate = _currentCoordinate.NewCoordinate(_currentDirection.XAxisStepSize, _currentDirection.YAxisStepSize);
        }

        public void TurnRight()
        {
            _currentDirection = _currentDirection.Right();
        }

        public void TurnLeft()
        {
            _currentDirection = _currentDirection.Left();
        }
    }
}